### PR TITLE
Java CDK: Automated Publish Workflow

### PR DESCRIPTION
**Pull request by aaronsteers**
This adds a publish workflow for the CDK.

Usage guidance is provided at the top of the workflow file:

```
# Usage: This workflow can be invoked manually or by a slash command.
#
# To invoke via GitHub UI, go to Actions tab, select the workflow, and click "Run workflow".
#
# To invoke via slash command, use the following syntax in a comment on a PR:
#    /publish-java-cdk                # Run with the defaults (dry-run=false, force=false)
#    /publish-java-cdk dry-run=true   # Run in dry-run mode (no-op)
#    /publish-java-cdk force=true     # Force-publish if needing to replace an already published version
```

Note: The file was created from a copy of the Python workflow. To see the fundamental diff from the Python version to the Java version, please use the GitHub commit selector to view the 2nd and 3rd commits:

- https://github.com/airbytehq/airbyte/pull/30674/files/125ae999d7638c772987d6df5b8946b59905fd9e..4f963f73a631385e0d07e6317e8aa5873e95a5ed#diff-cadf4aa9c9361dcfe3c2fea7dfc388a68d7dfd1f2d47b640290e4cbe0855774a

### CODE REVIEW

File: .github/workflows/publish-cdk-command-manually.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Publish CDK Manually
+name: Publish Python CDK Manually
```
Comment (aaronsteers) [inline .github/workflows/publish-cdk-command-manually.yml:1]: In this original file, I added "Python" everywhere to disambiguate. (Should be only cosmetic changes.)

**Comment by aaronsteers** [comment]
Pipeline appears to be working well, executes in 3min.

Note that this is 'dry run' mode that only does MavenLocal publish.

image

File: .github/workflows/publish-java-cdk-command.yml [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+          fi
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: "zulu"
+          java-version: "17"
+      - name: Build Java CDK
+        run: ./gradlew --no-daemon :airbyte-cdk:java:airbyte-cdk:build
+      - name: Publish Java Modules to LocalMaven (Dry-Run)
+        run: ./gradlew --no-daemon :airbyte-cdk:java:airbyte-cdk:publishToMavenLocal
```
Comment (bnchrch) [inline .github/workflows/publish-java-cdk-command.yml:141]: @aaronsteers is `publishToMavenLocal` a prereq to `publish` or can we skip this step if `DRY_RUN` is false?

Comment (aaronsteers) [reply .github/workflows/publish-java-cdk-command.yml:141]: We could absolutely skip if we want to. I didn't see any harm to it running in both scenarios though. If you'd like, I can toggle it off when a dry-run is not set

Comment (bnchrch) [reply .github/workflows/publish-java-cdk-command.yml:141]: Ah gotcha! Then my only minor nit pick is we should skip this step.

My thinking behind that is
1. we cant really debug these github actions without heavily modifiying these workflow files, and even then its a nightmare
2. would give us the smallest speed savings when publishing 🏎️

**Review by bnchrch — APPROVED** [review]
As far as github actions go this sees great to me :) 

and as a java novice nothing is jumping out as an issue

File: .github/workflows/publish-java-cdk-command.yml (hunk 2 above)
Comment (aaronsteers) [reply .github/workflows/publish-java-cdk-command.yml:142]: Done 👍

**Pull request merged**